deploy/kubernetes/operator/pkg/webhook/inspector/pod.go

```diff
@@ -100,8 +102,33 @@ func (i *inspector) validateDeletingShuffleServer(ar *admissionv1.AdmissionRevie
 }
 
 func (i *inspector) ifShuffleServerCanBeDeleted(rss *unifflev1alpha1.RemoteShuffleService, pod *corev1.Pod) bool {
-	if rss.Status.Phase != unifflev1alpha1.RSSUpgrading && rss.Status.Phase != unifflev1alpha1.RSSTerminating &&
-		pod.Status.Phase != corev1.PodFailed {
+	if rss.Spec.ShuffleServer.Autoscaler.Enable {
+		shuffleServerName := shuffleserver.GenerateName(rss)
+		hpa, err := i.hpaLister.HorizontalPodAutoscalers(rss.Namespace).Get(shuffleServerName)
+		if err != nil {
+			klog.Errorf("failed to get hpa when hpa is enabled: %v", err)
+			return false
+		}
+		var sts *appsv1.StatefulSet
+		sts, err = i.stsLister.StatefulSets(rss.Namespace).Get(shuffleServerName)
+		if err != nil {
+			klog.Errorf("failed to get sts when hpa is enabled: %v", err)
+			return false
+		}
+		// In this case, we have enabled HPA and HPA triggers the StatefulSet shrink, so we should allow deletion.
+		if *sts.Spec.Replicas == hpa.Status.DesiredReplicas && *sts.Spec.Replicas < sts.Status.Replicas {
```

Comment:
This may be triggered by the administrator, but the administrator is not advised to change the number of shuffle server copies after the HPA object is enabled.